models/issue_comment.go

```diff
@@ -571,6 +603,45 @@ func CreateRefComment(doer *User, repo *Repository, issue *Issue, content, commi
 	return err
 }
 
+// ClearPullPushComent clear all the push commit on issue since fore push
+func ClearPullPushComent(issue *Issue) error {
```

Comment:
I don't think this right. Suppose there are two commits that have previously been pushed to a PR:
```
sha1 newest (my parent is sha2)
sha2 older (my parent is something else)
```
Suppose `sha1` is removed via a force push, but `sha2` remains. `sha2`'s comment should not be deleted. Even if we recreate it later, the new comment will have a different timestamp than the old deleted one, which affects the in order in which comments are listed.

Reply:
Since it's a force push the old commit maybe changed also. We cannot detect that. And it seems github also always delete all previous commit comments.

Reply:
We can detect it. The commits whose comments need to be deleted are exactly the commits listed in the output from the `git rev-list ...` command.

Reply:
We can detect it. The commits whose comments need to be deleted are exactly the commits listed in the output from the `git rev-list ...` command.

Reply:
@ethantkoenig could you give me some example pull request on github the commit comments are not deleted when force push?

Reply:
![image](https://user-images.githubusercontent.com/8990880/32148844-54b8d446-bcb9-11e7-9c21-03a5507699dc.png)

![image](https://user-images.githubusercontent.com/8990880/32148852-6d15f30c-bcb9-11e7-8232-877190ab1652.png)